src/services/legal-services.ts

```diff
@@ -1,76 +1,87 @@
         // otherwise expect response to be a single filing
         if (response?.data?.filing) filing = response.data.filing
-        const filingName = filing?.header?.name
+
+        const filingName = filing?.header?.name as FilingTypes
         const filingId = +filing?.header?.filingId || 0
 
         if (!filing || !filingName || !filingId) {
           throw new Error('Invalid API response')
         }
 
-        switch (filingName) {
-          case FilingTypes.INCORPORATION_APPLICATION:
-            return this.formatEmptyIncorporationApplication(filing)
-          case FilingTypes.REGISTRATION:
-            return this.formatEmptyRegistration(filing)
-          default:
-            throw new Error('Invalid filing name')
-        }
+        return filing
       })
   }
 
   /**
-   * Fetches a draft dissolution filing.
-   * @param id the business id
-   * @returns a promise to return the draft dissolution filing, else exception
+   * Fetches the first task.
+   * This is expected to be a draft Dissolution or Restoration.
+   * @param businessId the business identifier
+   * @returns a promise to return the draft filing, else exception
    */
-  static async fetchDraftDissolution (id: string): Promise<DissolutionFilingIF> {
-    // get the draft filing from the tasks endpoint
-    const url = `businesses/${id}/tasks`
+  // eslint-disable-next-line max-len
+  static async fetchFirstTask (businessId: string): Promise<DissolutionFilingIF | RestorationFilingIF> {
+    const url = `businesses/${businessId}/tasks`
     return axios.get(url)
       .then(response => {
-        // look for dissolution task
-        const filing = response?.data?.tasks
-          // eslint-disable-next-line no-prototype-builtins
-          ?.find(x => x.task.filing.hasOwnProperty(FilingTypes.VOLUNTARY_DISSOLUTION))?.task.filing
-        const filingName = filing?.header?.name
+        const filing = response?.data?.tasks?.[0]?.task.filing
+        const filingName = filing?.header?.name as FilingTypes
         const filingId = +filing?.header?.filingId || 0
 
-        if (!filing || filingName !== FilingTypes.VOLUNTARY_DISSOLUTION || !filingId) {
+        if (!filing || !filingName || !filingId) {
           throw new Error('Invalid API response')
         }
 
         return filing
       })
   }
 
+  /**
+   * Fetches the specified filing.
+   * @param businessId the business identifier
+   * @param filingId the filing id
+   * @returns a promise to return the filing, else exception
+   */
+  static async fetchFiling (businessId: string, filingId: number): Promise<any> {
+    // get the draft filing from the tasks endpoint
+    const url = `businesses/${businessId}/filings/${filingId}`
+    return axios.get(url)
+      .then(response => {
+        const filing = response?.data?.filing
+        if (!filing) throw new Error('Invalid API response')
+        return filing
+      })
+  }
```

Comment:
We're not using this yet but I expect/hope we will when we pass the filing id from Filings UI to Create UI. Re: https://app.zenhub.com/workspaces/entities-team-space-6143567664fb320019b81f39/issues/gh/bcgov/entity/14838